src/main/java/de/tum/cit/aet/artemis/programming/domain/ProgrammingExercise.java

```diff
@@ -90,6 +90,9 @@ public String getType() {
     @Column(name = "max_static_code_analysis_penalty", table = "programming_exercise_details")
     private Integer maxStaticCodeAnalysisPenalty;
 
+    @Column(name = "allow_branching", table = "programming_exercise_details")
```

Comment:
we already have a boolean flag in `ProgrammingExerciseBuildConfig` (Java) and `programming_exercise_build_config` (database table). Can you please reuse this one so we avoid declaring a new field here that is fetched all the time without being needed in most places?